lib/scanner.l

```diff
@@ -167,16 +168,31 @@ include_open      ^[ \t]*@include[ \t]+\"
 {float}           { yylval->fval = atof(yytext); return(TOK_FLOAT); }
 {integer}         {
                     long long llval;
-                    llval = atoll(yytext);
+                    char *endptr;
+                    int errsave = errno;
+                    errno = 0;
+                    llval = strtoll(yytext, &endptr, 0);	/* base 10 or base 8 */
+                    if(*endptr || errno)
+                    {
+                      errno = 0;
+                      return(TOK_ERROR);	/* some error occured ... */
+                    }
+                    errno = errsave;
+                    if((*yytext=='0') && (*(yytext+1)!='\0'))
+                    {   /* it's octal... so INT we go */
+                      yylval->ival=(int)(llval);
```

Comment:
On 12/29/2015 11:41 PM, hyperrealm wrote:

Done.
New commit: 
https://github.com/jltallon/libconfig/commit/5db0a60a5e907bf7afa7ebb95b59d30551c30453


     Just tell me if you prefer me to rebase this or the other one... or 
neither.